### DISCUSSION TITLE
Different COlumns for Different Signal

### DISCUSSION THREAD

**Discussion by waqarayub68** [blue-yonder/tsfresh#882] [Q&A]
I have a set of pressure signals (15 signals) when I applied TSFRESH it gives me set of features but the issue is for each signal extracted features are different means to say different column names due to which I have lot of missing values.

Is there any way to extract features that are relevant to all the signals.

Thanks and Regards

**Comment by nils-braun** [comment]
Hi! 
What do you mean with different signals? Do you mean you have a multi-class classification problem? In this case, there is a setting to turn on multi-class mode in the `select_features` call, see [here](https://tsfresh.readthedocs.io/en/latest/api/tsfresh.feature_selection.html#tsfresh.feature_selection.selection.select_features).

**Reply by waqarayub68** [reply]
I mean to say I have Multiple Water Pressure Signals I want to cluster those signals into different groups (Just Like Simple Dataframe in which we have multiple rows I have multiple signals ). For clustering I need features for that I use TSFRESH but for some signals it gives different features and for some it gives different features which causes huge amount of missing values.

**Reply by nils-braun** [reply]
You can try to put them all into a single data frame, give each signal a unique ID and let tsfresh extract features on all of them at the same time. This will only give you the features which are relevant to all. The column names from tsfresh are derived from the input column names
